Objects/bytes_methods.c

```diff
@@ -582,10 +627,57 @@ _Py_bytes_index(const char *str, Py_ssize_t len, PyObject *sub,
 }
 
 PyObject *
-_Py_bytes_rfind(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_rfind(const char *str, Py_ssize_t len, PyObject *subobj,
                 Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "rfind", sub, start, end, -1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in chunks
+        Py_ssize_t cur_end = end;
+        for (; result == -1 && cur_end >= start; cur_end -= RFIND_CHUNK_SIZE) {
+            Py_ssize_t cur_start = cur_end - RFIND_CHUNK_SIZE + 1;
+            if (cur_start < start) {
+                cur_start = start;
+            }
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sub_len;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
+                    sub_len = 1;
+                }
+                else if (PyObject_GetBuffer(subseq, &subbuf,
+                                            PyBUF_SIMPLE) != 0)
+                {
+                    return NULL;
+                }
+                else {
+                    sub_len = subbuf.len;
+                }
+                Py_ssize_t sub_end = cur_end + sub_len;
```

Comment:
Here need to do: `sub_start = cur_start - sub_len`
Otherwise, say if `len(substr) == 3` and `chunk_size == 2`, it will fail to extend properly.

Reply:
It will match it in the next chunk.

```python
string = "abc"
subs = "abc",
slice0 = "c"    # no match
slice1 = "abc"  # match
```

The start *must* be the same for all substrings, otherwise bugs will occur:

```python
string = "abcdef"
subs = "abc", "b"
slice_a = "abc"    # match
slice_b = "c"      # no match (bug)
```

Reply:
That is a very good point. What is the definition of `rfind`? Is it finding the highest index of beginning or end of the string? What does `re` do?

E.g.:
```
s = 'aba'
sub = ('aba', 'b')
# which one is last?
i = re.finditer(r'(?s:.*)(aba)|b', 'aba')
next(i)    # Match: 'aba'
```
With current logic, 'aba' might not fit into the current window, it not being extended forwards (in reverse sense) and find 'b' first. Or is it what is wanted? I think need to check how other tools define this.

Reply:
I guess It is a first match going backwards, so I guess 'b' is first. In this case, it is correct as it is now.

Reply:
Regex:

```re
^[\s\S]*(?P<sub>aba|b)`
```

Matches `b` as expected, try it out here: https://regex101.com